include/commissioner/network_diag_data.hpp

```diff
@@ -0,0 +1,170 @@
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file defines the types of Thread Network Diagnostic TLVs used for network diagnostics.
+ */
+
+#ifndef OT_COMM_NETWORK_DIAG_DATA_HPP_
+#define OT_COMM_NETWORK_DIAG_DATA_HPP_
+
+#include <cstddef>
+#include <cstdint>
+#include <stdbool.h>
+#include <string>
+#include <vector>
+
+#include "defines.hpp"
+#include "error.hpp"
+
+namespace ot {
+
+namespace commissioner {
+
+/**
+ * @brief Mode Data
+ */
+struct ModeData
+{
+    bool mIsRxOnWhenIdleMode          = false;
+    bool mIsMtd                       = false;
+    bool mIsStableNetworkDataRequired = false;
+};
+
+/**
+ * @brief Child Entry in Child Table
+ */
+struct ChildTableEntry
+{
+    uint32_t mTimeout             = 0;
+    uint8_t  mIncomingLinkQuality = 0;
+    uint16_t mChildId             = 0;
+    ModeData mModeData;
+};
+
+/**
+ * @brief Leader Data
+ */
+struct LeaderData
+{
+    uint32_t mPartitionId       = 0;
+    uint8_t  mWeighting         = 0;
+    uint8_t  mDataVersion       = 0;
+    uint8_t  mStableDataVersion = 0;
+    uint8_t  mRouterId          = 0;
+};
+
+/**
+ * @brief Route Data Entry of RouteData in Route64
+ */
+struct RouteDataEntry
+{
+    uint8_t mRouterId            = 0;
+    uint8_t mOutgoingLinkQuality = 0;
+    uint8_t mIncomingLinkQuality = 0;
+    uint8_t mRouteCost           = 0;
+};
+
+/**
+ * @brief Route64 Data
+ */
+struct Route64
+{
+    uint8_t                     mIdSequence = 0;
+    ByteArray                   mMask;
+    std::vector<RouteDataEntry> mRouteData;
+};
+
+/**
+ * @brief Child IPv6 Address Info
+ */
+struct ChildIpv6AddressInfo
+{
+    uint16_t                 mRloc16 = 0;
+    std::vector<std::string> mAddrs;
+};
+
+/**
+ * @brief MAC Counters
+ */
+struct MacCounters
+{
+    uint32_t mIfInUnknownProtos  = 0;
+    uint32_t mIfInErrors         = 0;
+    uint32_t mIfOutErrors        = 0;
+    uint32_t mIfInUcastPkts      = 0;
+    uint32_t mIfInBroadcastPkts  = 0;
+    uint32_t mIfInDiscards       = 0;
+    uint32_t mIfOutUcastPkts     = 0;
+    uint32_t mIfOutBroadcastPkts = 0;
+    uint32_t mIfOutDiscards      = 0;
+};
+
+/**
+ * @brief network diagnostic data in TMF
+ *
+ * Each data field of Diagnostic TLVs is optional. The field is
+ * meaningful only when associative PresentFlags is included in
+ * `mPresentFlags`.
+ */
+struct NetDiagData
+{
+    ModeData                          mMode;
+    ByteArray                         mEui64;
+    ByteArray                         mExtMacAddress;
+    uint16_t                          mMacAddress = 0;
+    Route64                           mRoute64;
+    LeaderData                        mLeaderData;
+    MacCounters                       mMacCounters;
+    std::vector<std::string>          mAddrs;
```

Comment:
Some place we use "Addr" while some place we use "Address", it doesn't matter which one is used, but it would be better to make it consistent. Looking at existing code, "Addr" is always used for addresses. Can we update the code in PR to follow the existing naming? Thanks

Reply:
I see, I changed `addrs` to `address`

Reply:
No, if you look at include/commissioner/commissioner.hpp, "Addr" is used but not "Address"